app/components/search.tsx

```diff
@@ -85,8 +86,26 @@ export default function Search({
   const results = tfFinishedLoadingRef.current ? searchResults : baselineSearchResults
   const model = tfFinishedLoadingRef.current ? 'tensorflow' : 'plaintext'
 
+  const hideSearchResults = () => setShowResults(false);
+  const handleBlur = (e) => {
+    // React will recreate the whole search widget if `showResults` changes.
+    // `true -> true` also counts as a change, so make sure that it only
+    // gets mutated when it really should be opened/closed
+    const focusedOnResult = e.relatedTarget?.classList.contains('result-item');
+    if (!focusedOnResult) {
+      hideSearchResults();
+    }
+  }
+  const onQuestionAdded = (title) => {
+      alert('Your question "' + title + '" was added. It might take a while for it to be shown here, as it has yet to be answered.');
+      hideSearchResults();
+  }
+
   return (
-    <div>
+    <div
+      onFocus={() => setShowResults(true)}
+      onBlur={handleBlur}
```

Comment:
this has to be here for focus changes in the suggested answers to be handled properly - otherwise it would never close once the input loses focus